### PR TITLE
Microsoft.Billing: Gtm ImportRequest swagger.

### PR DESCRIPTION
**Pull request by SamhitaK6**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by anuchandy — CHANGES_REQUESTED** [review]
Please see the review comments. Also there are example validation [errors](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/439908791), can you fix them.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json [hunk 1]
```diff
@@ -0,0 +1,1371 @@
...
+        },
+        "billingAccountId": {
+          "type": "string",
+          "description": "Target billing account Id.",
+          "readOnly": true
+        },
+        "devTestUsageContext": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Target usage ontext for dev-test  susbcriptions."
```
Comment (KrisBash) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: missing a "c" in context

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json [hunk 2]
```diff
@@ -0,0 +1,1371 @@
...
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/initiateImportRequestProperties"
+        }
+      }
+    },
+    "initiateImportRequestProperties": {
+      "type": "object",
+      "description": "Request parameters to initiate import request.",
+      "properties": {
+        "devTestUsageContext": {
```
Comment (KrisBash) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: Does the user understand what a "context" is? Is this an arbitrary string or is some server-side validation occur?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json [hunk 3]
```diff
@@ -0,0 +1,1371 @@
...
+          "readOnly": true,
+          "description": "Email Id of initiator of import request."
+        },
+        "canceledBy": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Email Id who user canceled the import request."
+        },
+        "lastModifiedTime": {
+          "type": "string",
```
Comment (KrisBash) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: string or date-time?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json [hunk 4]
```diff
@@ -0,0 +1,1371 @@
...
+          }
+        }
+      }
+    },
+    "recipientImportRequestProperties": {
+      "description": "Import request Details.",
+      "type": "object",
+      "properties": {
+        "creationTime": {
+          "type": "string",
```
Comment (KrisBash) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: string or date-time?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json [hunk 5]
```diff
@@ -0,0 +1,1371 @@
...
+          "description": "Import request creation time.",
+          "readOnly": true
+        },
+        "expirationTime": {
+          "type": "string",
```
Comment (KrisBash) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: string or date-time?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json [hunk 6]
```diff
@@ -0,0 +1,1371 @@
...
+        "allowedSubscriptionType": {
+          "type": "array",
+          "readOnly": true,
+          "description": "Type of subscriptions that can be imported.",
+          "items": {
+            "$ref": "#/definitions/subscriptionType"
+          }
+        },
+        "importRequestStatus": {
+          "type": "string",
```
Comment (KrisBash) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: Is this a finite list of status values? Can it be an enum?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json [hunk 7]
```diff
@@ -0,0 +1,1371 @@
...
+          "readOnly": true,
+          "description": "Email Id of initiator of import request."
+        },
+        "canceledBy": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Email Id who user canceled the import request."
+        },
+        "lastModifiedTime": {
+          "type": "string",
```
Comment (KrisBash) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: date-time?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json [hunk 8]
```diff
@@ -0,0 +1,1371 @@
...
+          "readOnly": true,
+          "description": "Type of product being imported."
+        },
+        "productId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Id of product being imported."
+        },
+        "importRequestStatus": {
+          "type": "string",
```
Comment (KrisBash) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: can this be an enum?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json [hunk 9]
```diff
@@ -0,0 +1,1371 @@
...
+          },
+          {
+            "$ref": "#/parameters/initiateImportRequestParameter"
+          }
+        ],
+        "tags": [
+          "AIPController"
+        ],
+        "responses": {
+          "200": {
```
Comment (KrisBash) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: Shouldn't this be a 202? Based on the presence of a status field in the response payload, I assume that the request may succeed or fail after the response is returned. 'Accepted' seems to be a more accurate response in this case.

**Review by KrisBash — CHANGES_REQUESTED** [review]
In the future, it would be helpful to copy the prior API version to the directory as an initial commit so that subsequent changes are exposed in the diff. Please see comments.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json [hunk 10]
```diff
@@ -0,0 +1,1371 @@
...
+            "description": "Initiated import request details.",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/importRequestDetails"
+              }
+            }
```
[the remark this answers is not in the document]
Comment (SamhitaK6) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: Fixed it.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json (hunk 2 above)
Comment (SamhitaK6) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: User understands what context is. Also there are few validations for it on server side.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json (hunk 9 above)
Comment (SamhitaK6) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: Import request operations defined in this swagger by themselves are not long running.
Like initiating the import request is not running operation, though actual import of product might take longer. So to track this import of product status there is status field provided in response.

**Review by KrisBash — APPROVED** [review]
LGTM for ARM

**Review by KrisBash — APPROVED** [review]
LGTM

**Comment by anuchandy** [comment]
@SamhitaK6 let me know once this PR is finalized. Is this api version deployed to production?

**Pull request closed without merging**